### PR TITLE
Bouncy platform Functionality 

### PR DESCRIPTION
**Pull request by MichaelSam05**
## Description
What changes does this PR include? Is there anything that affects other features that people should be aware of?
- Added bounce platform and trampoline functionality- auto bounces player in respective direction when contact is made
- Added player bounce related constants that determine: how high player goes in the y direction and how far player goes in the x direction.
- Added getter and setter methods for player velocity in PlayerController
- Added private accessor variables for trampoline script in PlayerController

## Bug Fixes
- Fixed issue where bounce mechanic behaves incorrectly when player dashes into bouncy platform 
- Fixed issue where player couldn't run into trampoline and bounce but was force to jump on it instead
- Fixed issue where bounce platform changed the player's direction randomly when travelling left
- Moved bounce related functionality into separate scripts so that constant can be applied on an individual basis rather than globally via PlayerController script

## Demo Video
 
https://github.com/user-attachments/assets/33a1864c-4e77-408f-9f14-9f7dca133ec2

## Checklist
- [x] This code builds and runs
- [x] All public classes and methods are documented
- [x] Hard-to-understand areas of the code are documented
- [x] Self-review done

### CODE REVIEW

File: Assets/Scenes/TemplateScene.unity
Comment (AdenC123) [inline Assets/Scenes/TemplateScene.unity:1]: Template scene should be unchanged, can u copy this over to a new test scene and revert it

**Review by AdenC123 — CHANGES_REQUESTED** [review]
Excited for bouncy platforms!! Added a couple organization things to change before merging. Just to check, is the intent for the trampoline prefab to always launch the player up and for the bouncy platform to bounce the player off in the opposite direction?

3 bigger requests:
1. For the trampoline prefab: right now it is pretty small and hard to notice, so the size should be increased. Also, it only launches the player when you land on it from the air- it would be nice to also launch up if you walk into it. Take a look at Sparsh's drawings for the tutorial level and see if the trampoline physics will work with that layout.
2. For the bouncy platform prefab: it should have a sprite (maybe a rubber-looking rectangle thing?) Also, the launch was pretty inconsistent when messing around with it- sometimes it gave me lots of height and sometimes only launched me horizontally, and one time it switched my direction to the right when going to the left. Added a couple ideas to improve consistency, but you know the bounce physics better than me so experiment with whatever ideas you have! Happy to work together on physics stuff as well.
3. We should be able to set properties like the bounce force and arc multiplier on each individual trampoline, so we can have multiple trampolines in the same level that act differently. You would add a script to the prefabs that has the properties you want, then load them into private variables in the player controller during OnCollisionEnter. It doesn't look like we will need multiple trampoline configs for the first level though, so ok to leave this for a separate PR later.

**Comment by MichaelSam05** [comment]
Ok thanks for the feedback will adjust the code as soon as I can and let you know if I run into any issues

**Comment by AdenC123** [comment]
Thx, lmk anytime if u have any questions

**Comment by MichaelSam05** [comment]
Hey Aden tried my best to fix some of the issues related to the bouncy platform. 
- The random direction changes no longer occur
- I figured out the source of the inconsistency when bouncing and fixed that. 
Wasn't able to make it so that the player could run into the platform and bounce automatically and I wasn't able to move the variables to another script yet as stuff began to break and I know we are pressed for time also I'm not versed on making sprites for objects so that still has to be done.

**Comment by kaisoapbox** [comment]
I don't think we want this PR closed, it sounds like we wanted to potentially merge it down the line; did you want it as a draft or something?

**Comment by MichaelSam05** [comment]
> I don't think we want this PR closed, it sounds like we wanted to potentially merge it down the line; did you want it as a draft or something?

My bad I was working on it last night and meant to open a new PR but fell asleep. Fixed the bugs I had just need to do some more testing to make sure there're gone. Will edit this PR when i'm done to reflect those changes

**Review by kaisoapbox — CHANGES_REQUESTED** [review]
not a full code review, im leaving that to @AdenC123 but a few things:
1. we gotta run the reformatter/code linter cos there's a bunch of random stuff out of place
2. there's some weird bugs (features?) relating to edge cases of the bouncy platform collision, namely:
* if you land right on the corner you bounce away from the corner (imo this could be a feature/tech and not a bug)
* if you hit an angled bouncy platform shortly after touching the ground, you sometimes reflect back flat along the ground instead of bouncing up at an angle (this one feels more like a bug imo)
4. some small tweaks to the trampoline collider size + impact force to make the bounce work better are needed i think. currently the values i like are edge radius 0 (instead of 1), x size 1 (instead of 2), and x bounce force 5 (instead of 10 or smth)

**Review by AdenC123 — CHANGES_REQUESTED** [review]
The physics feel much better here! The demo level is fun to bounce around in, excited to see it in the main levels.

Kai gave some good feedback on hitbox size / parameters. Code wise, there is a few changes needed- mainly, physics code should be moved to PlayerController. I gave some more details in the comments, let me know if you have any questions about it.

Also, there is some non-standard names and formatting- if you open the code in Rider, there should not be any warnings about names (ideally no warnings at all). Mainly, classes should be in PascalCase and private variables should start with an underscore. Rider will rename everything for you in the context menu. To fix formatting automatically, you can run Rider's code formatter- just make sure to disable the formatter around the private serialized fields if you don't want them to look ugly (check PlayerController for an example).

**Comment by MichaelSam05** [comment]
@kaisoapbox I THINK the weird edge case bug is gone since I move the bounce physics back into playerController. My best guess for why it happened was that before, in the reflection calculations, i grabbed the player object which is a capsule collider but now i grab the platform itself which is a box collider: perhaps the curved shape of the player induced those edge cases?

**Review by AdenC123 — APPROVED** [review]
LGTM! I just ran Rider's formatter on the player controller, I'll look into getting our code automatically formatted on remote.

**Review by kaisoapbox — APPROVED** [review]

**Comment by kaisoapbox** [comment]
yep, i dont think i've managed to reproduce the bounce since the fix so it seems good to me! will merge!

**Pull request merged**